### PR TITLE
Add Repo definitions for openSUSE Leap 42.1

### PR DESCRIPTION
**Pull request by DimStar77**
https://progress.opensuse.org/issues/8942

A lot of repos in Additional are not yet enabled.. I'll get in touch with the maintainers to see if they want them added...

### CODE REVIEW

File: YaST/Repos/_openSUSE_Leap_42.1_Default.xml.in [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        <_summary>Repository for official non free updates to openSUSE Leap 42.1</_summary>
+        <_description>In this repository you find security and maintenance updates to openSUSE Leap 42.1.</_description>
+        <url>http://download.opensuse.org/update/leap/42.1-non-oss/</url>
+      </repository>
+            <repository recommended="false" format="yast">
+                <_name>Update Repository (DEBUG)</_name>
+                <_summary>Update repository of openSUSE Leap 42.1 debuginfo packages</_summary>
+                <_description>This repository is useful for those that want to debug applications on openSUSE Leap 42.1. For experts only.</_description>
+                <url>http://download.opensuse.org/debug/update/leap/42.1/</url>
+            </repository>
```
Comment (coolo) [inline YaST/Repos/_openSUSE_Leap_42.1_Default.xml.in]: that's not where it is nowadays - the update repos moved

File: YaST/Repos/_openSUSE_Leap_42.1_Default.xml.in [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+		 </_description>
+		<url>http://download.opensuse.org/distribution/leap/42.1/repo/oss/</url>
+	    </repository>
+	    <repository recommended="true" format="yast">
+		<_name>Main Repository (NON-OSS)</_name>
+		<_summary>Non-Open Source Software Addon repository for openSUSE Leap 42.1</_summary>
+		<_description>The official openSUSE Leap 42.1 repository for all
+		Non-Open Source Software maintained by the openSUSE team,
+		including Opera, Java, Flash, and more.</_description>
+		<url>http://download.opensuse.org/distribution/leap/42.1/repo/non-oss/</url>
```
Comment (coolo) [inline YaST/Repos/_openSUSE_Leap_42.1_Default.xml.in]: java is oss (for a decade or so :)

File: YaST/Repos/openSUSE_Leap_42.1_Servers.xml.in [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+	<item>
+	    <link>http://download.opensuse.org/YaST/Repos/_openSUSE_Leap_42.1_Additional.xml</link>
+	    <official config:type="boolean">true</official>
+	</item>
+
+	<!-- Community links -->
+	<item>
+	    <link>http://opensuse-community.org/openSUSE_Leap_42.1_Community_Additional.xml</link>
+	    <official config:type="boolean">false</official>
```
Comment (coolo) [inline YaST/Repos/openSUSE_Leap_42.1_Servers.xml.in]: you need to comment that till it exists - or yast will barf

File: YaST/Repos/_openSUSE_Leap_42.1_Default.xml.in (hunk 1 above)
Comment (DimStar77) [inline YaST/Repos/_openSUSE_Leap_42.1_Default.xml.in]: Actually just pushed a fix here - there is only the debug/update missing

File: YaST/Repos/_openSUSE_Leap_42.1_Default.xml.in (hunk 2 above)
Comment (DimStar77) [inline YaST/Repos/_openSUSE_Leap_42.1_Default.xml.in]: Removed Java, mentioned Steam in place (which is something users might be trying to find after all and we have it in the repo)

**Pull request merged**